### PR TITLE
update on the snews tier publisher

### PR DESCRIPTION
**Pull request by KaraMelih**
- It allows for environment setting (this is needed if the user doesn't pass detector name, it tries to fetch from env)
- can read from json file (simple `from_json()` method, parses and passes this data)
- arbitrary number of key-value pairs; `init` takes `env_file=None, **kwargs` and anything can be passed, it is looked in the `snews_pt_utils._tier_decider` and both the messages and which tiers found is returned
- it keeps the messages and tiers as attributes; the `messages` and `tiernames` (see above) are appended as attribute thus;

```
from SNEWS_PT.snews_pub import SNEWSTiersPublisher
snews = SNEWSTiersPublisher(detector_name='KamLAND', timing_series=['4:31:05:565', '4:31:05:575', '4:31:05:585'],
                                                  p_val=0.000007, p_values=[0.000007, 0.000004], machine_time='22/02/08 4:31:08:565')
```
can be further investigated before submitting a message i.e.
```
print(snews.messages)
print(snews.tiernames)
```
and finally `snews.send_to_snews()` submits the message.

Alternatively;
```
from SNEWS_PT.snews_pub import SNEWSTiersPublisher
snews = SNEWSTiersPublisher().from_json('myjsonfile.json')
snews.send_to_snews()
```

### CODE REVIEW

**Comment by KaraMelih** [comment]
@Storreslara Today we were working on your branch which is why I created this PR for your branch. Once a PR is created it is easier to compare the changes. Could you please have a look and maybe we can go over the lines by commenting?

File: SNEWS_PT/snews_pub.py [hunk 1]
```diff
@@ -76,91 +77,42 @@ def display_message(self, message):
...
+        snews_pt_utils.set_env(env_file)
+        self.args_dict = dict(**kwargs)
+        # if is_pre_sn not given, it is False. If name not given fetch from env
+        self.args_dict["is_pre_sn"] = kwargs.get("is_pre_sn", False)
+        self.args_dict['detector_name'] = kwargs.get("detector_name", os.getenv('DETECTOR_NAME'))
+        self.schema = Message_Schema(detector_key=self.args_dict['detector_name'], is_pre_sn=kwargs.get("is_pre_sn", False))
+        self.messages, self.tiernames = snews_pt_utils._tier_decider(self.args_dict)
+
+        # if we want to have all of them as attributes,
+        # for key, value in kwargs.items():
```
Comment (KaraMelih) [inline SNEWS_PT/snews_pub.py:104]: here we can actually add everything as individual attributes this is equal to e.g.
`self.p_values = args['p_values']`
but we already have the `self.args_dict`, and also individual formatted messages (and what tier they correspond to)

File: SNEWS_PT/snews_pt_utils.py [hunk 2]
```diff
@@ -266,126 +265,56 @@ def heartbeat_data(machine_time=None,
...
-#             click.secho(f'\t\t> {name}', fg='cyan')
-#     return tiers_unique, names_unique
-
 
 def _tier_decider(data:dict) -> tuple:
     """ Decide on the tier(s) or commands (Heartbeat/Retraction)
         Based on the content of the message
 
     """
-    from .snews_pub import SNEWSTiers
```
Comment (Storreslara) [inline SNEWS_PT/snews_pt_utils.py:341]: Wouldn't it be best to have the imports  at the top of the script ?

File: SNEWS_PT/snews_pub.py [hunk 3]
```diff
@@ -61,6 +61,7 @@ def send(self, messages):
 
         """
         for message in messages:
+            print(message)
```
Comment (Storreslara) [inline SNEWS_PT/snews_pub.py:64]: we don't need a `print` statement since we already have `display_message()`

**Review by Storreslara — COMMENTED** [review]
Just commented on little things, great work

File: SNEWS_PT/snews_pt_utils.py (hunk 2 above)
Comment (KaraMelih) [reply SNEWS_PT/snews_pt_utils.py:341]: This was in the previous version, in the latest commit I completely removed `SNEWSTiers` import as we don't need it.
Little more on that; I was looking at its signature and then assign the remaining keys to be `meta` but now that `SNEWSTierPublisher` doesn't take arguments (instead takes key-value pairs as kwargs) it doesn't have a clear signature, thus here I added a `valid_keys` list. Now, if the input keys are not in this list, they are assigned as `meta`.

In `SNEWSTiersPublisher` user can still see the relevant keys in docstring and also on message schema

File: SNEWS_PT/snews_pub.py (hunk 3 above)
Comment (KaraMelih) [reply SNEWS_PT/snews_pub.py:64]: you are right, this is a relic from testing 😄 I'll get rid of it

Comment (KaraMelih) [reply SNEWS_PT/snews_pub.py]: done and committed 💯

**Comment by KaraMelih** [comment]
In this 5cb1ac37ff1d4d6533280d56f64a209d1b5ce2e4 commit, I changed the key `'nu_time'` to `'neutrino_time'` and also `'p_value'` to `'p_val'`

**Pull request merged**